### PR TITLE
[docs] Adding a new model documentation

### PR DESCRIPTION
**Pull request by vedanuj**
- Add a doc for adding a new model. Uses Concat BERT

### CODE REVIEW

File: website/docs/tutorials/concat_bert.md [hunk 1]
```diff
@@ -0,0 +1,262 @@
+---
+id: concat_bert
+title: 'Tutorial: Adding a model - Concat BERT'
+sidebar_label: Adding a model - Concat BERT
+---
+
+In this tutorial we will go through the step-by-step process of creating a new model using MMF. In this case, we will create a fusion model for the [Hateful Memes dataset](https://github.com/facebookresearch/mmf/tree/master/projects/hateful_memes).
+
+The fusion model that we will create concatenates embeddings from a text encoder and an image encoder and passes them through a two-layer classifier. MMF provides standard image and text encoders out of the box. For image encoder, we will use ResNet152 image encoder and for text encoder, we will use BERT Base Encoder.
```
Comment (apsdehal) [inline website/docs/tutorials/concat_bert.md:9]:
```suggestion
The fusion model that we will create concatenates embeddings from a text encoder and an image encoder and passes them through a two-layer classifier. MMF provides standard image and text encoders out of the box. For the image encoder, we will use the ResNet152 image encoder and for the text encoder, we will use the BERT-Base Encoder.
```

File: website/docs/tutorials/concat_bert.md [hunk 2]
```diff
@@ -0,0 +1,262 @@
+---
+id: concat_bert
+title: 'Tutorial: Adding a model - Concat BERT'
+sidebar_label: Adding a model - Concat BERT
+---
+
+In this tutorial we will go through the step-by-step process of creating a new model using MMF. In this case, we will create a fusion model for the [Hateful Memes dataset](https://github.com/facebookresearch/mmf/tree/master/projects/hateful_memes).
```
Comment (apsdehal) [inline website/docs/tutorials/concat_bert.md:7]:
```suggestion
In this tutorial, we will go through the step-by-step process of creating a new model using MMF. In this case, we will create a fusion model which we will train on the [Hateful Memes dataset](https://github.com/facebookresearch/mmf/tree/master/projects/hateful_memes).
```

File: website/docs/tutorials/concat_bert.md [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+
+The fusion model that we will create concatenates embeddings from a text encoder and an image encoder and passes them through a two-layer classifier. MMF provides standard image and text encoders out of the box. For image encoder, we will use ResNet152 image encoder and for text encoder, we will use BERT Base Encoder.
+
+
+## Installation and Preparing the dataset
+
+Follow the prerequisites for installation and dataset [here](https://github.com/facebookresearch/mmf/tree/master/projects/hateful_memes#prerequisites).
```
Comment (apsdehal) [inline website/docs/tutorials/concat_bert.md:14]: link the installation docs directly?

File: website/docs/tutorials/concat_bert.md [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+title: 'Tutorial: Adding a model - Concat BERT'
+sidebar_label: Adding a model - Concat BERT
+---
+
+In this tutorial we will go through the step-by-step process of creating a new model using MMF. In this case, we will create a fusion model for the [Hateful Memes dataset](https://github.com/facebookresearch/mmf/tree/master/projects/hateful_memes).
+
+The fusion model that we will create concatenates embeddings from a text encoder and an image encoder and passes them through a two-layer classifier. MMF provides standard image and text encoders out of the box. For image encoder, we will use ResNet152 image encoder and for text encoder, we will use BERT Base Encoder.
+
+
+## Installation and Preparing the dataset
```
Comment (apsdehal) [inline website/docs/tutorials/concat_bert.md:12]: Change this section to prerequisites?

File: website/docs/tutorials/concat_bert.md [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+
+Follow the prerequisites for installation and dataset [here](https://github.com/facebookresearch/mmf/tree/master/projects/hateful_memes#prerequisites).
+
+
+## Using MMF to build the model
+
+We will start building our model `ConcatBERT` using the various building blocks available in MMF. Helper builder methods like  `build_image_encoder` for building image encoders, `build_text_encoder` for building text encoders, `build_classifier_layer` for classifier layers etc take configurable params which are defined in the config we will create in the next section.
```
Comment (apsdehal) [inline website/docs/tutorials/concat_bert.md:19]:
```suggestion
We will start building our model `ConcatBERT` using the various building blocks available in MMF. The helper builder methods like  `build_image_encoder` for building image encoders, `build_text_encoder` for building text encoders, `build_classifier_layer` for classifier layers etc take configurable params which are defined in the config we will create in the next section.

Follow the code and read through the comments to understand the process.
```

File: website/docs/tutorials/concat_bert.md [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+  batch_size: 64
+  lr_scheduler: true
+  max_updates: 22000
+  early_stop:
+    criteria: hateful_memes/roc_auc
+    minimize: false
+
+```
+
+We include the `bert.yaml` config in this as we want to use BERT tokenizer for preprocessing our language data. With both the configs ready we are ready to launch training and evaluation using our model on Hateful Memes dataset. You can read more about the MMF’s configuration system [here](https://mmf.sh/docs/notes/configuration).
```
Comment (apsdehal) [inline website/docs/tutorials/concat_bert.md:247]:
```suggestion
We include the `bert.yaml` config in this as we want to use BERT tokenizer for preprocessing our language data. With both the configs ready we are ready to launch training and evaluation using our model on the Hateful Memes dataset. You can read more about the MMF’s configuration system [here](https://mmf.sh/docs/notes/configuration).
```

File: website/docs/tutorials/concat_bert.md [hunk 7]
```diff
@@ -0,0 +1,262 @@
...
+
+## Model Config
+
+We will now create the model config file with the params we used while creating the model and store the config in `configs/models/concat_bert/defaults.yaml`.
+
+```yaml
+
+model_config:
+  concat_bert:
+    # Either pretraining or classification
```
Comment (apsdehal) [inline website/docs/tutorials/concat_bert.md:160]: This seems misplaced.

File: website/docs/tutorials/concat_bert.md [hunk 8]
```diff
@@ -0,0 +1,262 @@
...
+        # MMF will automatically calculate loss
+        return output
+
+```
+
+
+The model’s forward method takes a `SampleList` and outputs a dict containing the logit scores predicted by the model. Different losses and metrics can be calculated on the scores output.
+
+
+We define two configs needed for our experiments: (i) model config for model defaults (ii) experiment config. Model config provides the defaults for model’s hyperparameters and experiment config defines and overrides the defaults needed for our particular experiment
```
Comment (apsdehal) [inline website/docs/tutorials/concat_bert.md:149]:
```suggestion
We define two configs needed for our experiments: (i) a model config for model's defaults and (ii) an experiment config. The model config provides the defaults for model’s hyperparameters and the experiment config defines and overrides the defaults needed for our particular experiment such as optimizer type, learning rate, maximum updates and batch size.
```

File: website/docs/tutorials/concat_bert.md [hunk 9]
```diff
@@ -0,0 +1,262 @@
...
+Now we are ready to train and evaluate our model with the experiment config we created in previous step.
+
+```bash
+mmf_run config="configs/experiments/concat_bert/defaults.yaml" \
+    model=concat_bert \
+    dataset=hateful_memes \
+    run_type=train_val
+```
+
+When training ends it will save a final model `concat_bert_final.pth` in the experiment folder under `./save` directory. The command will also generate validation scores after the training gets over.
```
Comment (apsdehal) [inline website/docs/tutorials/concat_bert.md:262]: Add link to general checkpointing tutorial here about prediction and evaluation.
